scex-core/src/test/scala/com/avsystem/scex/compiler/MultipleEvaluationsTest.scala

```diff
@@ -0,0 +1,29 @@
+package com.avsystem.scex.compiler
+
+import com.avsystem.scex.compiler.presentation.TypeCompletionPrefixTest.Root
+import com.avsystem.scex.japi.{DefaultJavaScexCompiler, JavaScexCompiler}
+import com.avsystem.scex.util.SimpleContext
+import org.scalatest.funsuite.AnyFunSuite
+
+final class MultipleEvaluationsTest extends AnyFunSuite with CompilationTest {
+  override protected def createCompiler: JavaScexCompiler = {
+    val settings = new ScexSettings
+    settings.classfileDirectory.value = "testClassfileCache"
+
+    // for some reason enabling this setting fixes validation
+    settings.noGetterAdapters.value = false
+    new DefaultJavaScexCompiler(settings)
+  }
+
+  private val completer = compiler.getCompleter[SimpleContext[Root], Any](
+    createProfile(Nil),
+    template = false,
+    header = "",
+  )
+
+  test("Multiple evaluations of invalid expressions should always fail") {
```

Comment:
Add test case with mixed correct and incorrect expressions

Also check corner case, when invalid expression is the last one before reaching `settings.resetAfterCount` and next expression is valid